### PR TITLE
Create documentation for working with configurations.

### PR DESCRIPTION
**Pull request by Xemiru**
Adds a guide for basic manipulation of configuration files. WIP.
Expands upon zml's branch in #145.

### CODE REVIEW

**Comment by Xemiru** [comment]
its good enough to be merged, though object mapping is missing. have to play around with it a little and figure out how it actually works.

File: source/plugin/advanced/config-in-depth/paths.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+.. code-block:: json
+
+    config.blockCheats.enabled=value
+
+... which is not the organized intended result we're looking for. Instead, what you -can- do is split the path by the separator character so it ends up as 3 separate arguments.
+
+.. code-block:: java
+
+    rootNode.getNode((Object[]) "config.blockCheats.enabled".split("\\."));
```
Comment (zml2008) [inline source/plugin/advanced/config-in-depth/paths.rst]: no. no. just no...

```java
rootNode.getNode("config", "blockCheats", "enabled")
```
varargs. learn what it is.don't give stupid examples.

Comment (Xemiru) [inline source/plugin/advanced/config-in-depth/paths.rst]: I did do prior testing and it does work with that method, and there isn't much danger from doing it that way, from the looks of it.

Comment (zml2008) [inline source/plugin/advanced/config-in-depth/paths.rst]: Only issue is that it's entirely the wrong way to go about it stylewise

**Comment by BlackGoat** [comment]
Any progress, or should I merge this? ... and can you please squash commits before I do?

**Comment by Xemiru** [comment]
I'll finalize it when I get home.

**Comment by Xemiru** [comment]
Squashed into 1 commit (ignoring the pull from master commit; i tried).
Object mapping remains unfinished.
Ready for merge.

**Comment by BlackGoat** [comment]
Being picky, it looks like a submodule has been included for some reason. The last change shows a tweak to src/sponge-docs-theme that doesn't look like it belongs here. Can you fix this before merge please? (or at least explain?)

**Comment by Xemiru** [comment]
.. huh.
think its cuz i was testing with a local server to see changes
ill fix that when im home

**Comment by Xemiru** [comment]
not sure how i'm supposed to rid of that subproject commit, git just tells me it doesn't exist locally

**Comment by Xemiru** [comment]
figured out removing the submodule.

**Pull request merged**